### DISCUSSION TITLE
Functions in local does not reach

### DISCUSSION THREAD

**Discussion by lerayj** [vercel/community#525] [Help] [answered]
Since I've upgraded vercel cli to 24.0.1, I cannot test my vercel functions locally. 
It used to work well before. I tried to downgrade the cli version, but I still go into this issue.
<img width="778" alt="Capture d’écran 2022-03-29 à 10 45 37" src="https://user-images.githubusercontent.com/5832419/160571718-7410f8e7-8ad2-4836-bf11-5b2491c15fe3.png">

Everything works in production - I guess there it is a port mapping issue with localhost? I have no explanation why it stopped working and I'm not able to get into my previous state by downgrading, mystery to me...

Does someone have any clue of to solve this? 

EDIT:
Running `npm run dev`, alias for `nuxt` command works properly.
<img width="840" alt="Capture d’écran 2022-03-29 à 17 01 58" src="https://user-images.githubusercontent.com/5832419/160642297-682d4eeb-6905-4759-b36d-5eee3e6e6821.png">

That is not the case with `vercel dev`, which have this config: 
![Capture d’écran 2022-03-29 à 16 56 06](https://user-images.githubusercontent.com/5832419/160640949-45a2e439-ee6d-407e-976b-a2d719276287.png)
When trying to access the `http://localhost:3000/`
<img width="814" alt="Capture d’écran 2022-03-29 à 17 03 14" src="https://user-images.githubusercontent.com/5832419/160642548-ef0b293d-cba1-4bc1-8809-e76cdb1f6c01.png">

Page received:
![Capture d’écran 2022-03-29 à 17 04 10](https://user-images.githubusercontent.com/5832419/160642738-4fa0aa8d-86dd-4954-b5f6-7af9d99de078.png)

**Comment by swarnava** [comment]
ECONNREFUSED means most probably you have not started your server on port 8080.

Take a look at the nodejs documentation [here](https://nodejs.org/api/errors.html#errors_common_system_errors)


>   ECONNREFUSED (Connection refused): No connection could be made because the target machine actively refused it. This usually results from trying to connect to a service that is inactive on the foreign host.

**Reply by lerayj** [reply]
Don't think that it's what happend.
The web app server run on port 3000 by default, I can make it run on 8080, this change nothing.
<img width="840" alt="Capture d’écran 2022-03-29 à 13 32 55" src="https://user-images.githubusercontent.com/5832419/160602181-dce12ae7-d237-443c-b568-b65e5382a7ac.png">


Each time I rerun the vercel dev seems to dynamically set a port - I dunno for instance on the screen what set the port 49757.
If there is another server emulating function endpoint, I guess the issue is here.

Or did I miss something on what your saying ?

**Reply by lerayj** [reply]
@swarnava I've added some screen on the question, does the issue make more sense ? Thanks for helping!

**Reply by swarnava** [reply]
It looks like you are already in touch with one of our support engineers. I have also noticed that this is happening only to a specific environment. I will recommend to update our support thread with all your findings and continue troubleshooting there. Thank you!

**Comment by lerayj** [comment, marked as answer]
For future, the issue is exaclty the same as  this [issue](https://github.com/vercel/vercel/issues/7306).

Removing ::1 from the hosts file make thinks work. Not sure it's node@17 related since even on node@15 or node@14 I had the issue.

I guess a better error messaging could help if it's happened to other in the future (and since it's pretty random) ?

Thanks anyway @swarnava!
